diff -up masscan-8746c78856ca9a4d20617dd5d234850a6136b0af/src/pixie-threads.h.secondary masscan-8746c78856ca9a4d20617dd5d234850a6136b0af/src/pixie-threads.h
--- masscan-8746c78856ca9a4d20617dd5d234850a6136b0af/src/pixie-threads.h.secondary	2014-01-21 14:28:36.000000000 +0100
+++ masscan-8746c78856ca9a4d20617dd5d234850a6136b0af/src/pixie-threads.h	2014-01-21 14:29:11.000000000 +0100
@@ -40,7 +40,7 @@ void pixie_locked_subtract_u32(unsigned
 #define pixie_locked_CAS32(dst, src, expected) __sync_bool_compare_and_swap((volatile int*)(dst),(int)expected,(int)src);
 #define pixie_locked_CAS64(dst, src, expected) __sync_bool_compare_and_swap((volatile long long int*)(dst),(long long int)expected,(long long int)src);
 
-#if defined(__arm__)
+#if ! (defined(__i386__) || defined(__x86_64__))
 #define rte_wmb() __sync_synchronize()
 #define rte_rmb() __sync_synchronize()
 #define rte_pause()
